src/contracts/pods/EigenPod.sol

```diff
@@ -194,34 +252,67 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         );
 
-        // Balance updates should not be "stale" (older than VERIFY_BALANCE_UPDATE_WINDOW_SECONDS)
-        require(
-            oracleTimestamp + VERIFY_BALANCE_UPDATE_WINDOW_SECONDS >= block.timestamp,
-            "EigenPod.verifyBalanceUpdates: specified timestamp is too far in past"
-        );
+        /// TODO: I got rid of this check, because proving against an old checkpoint requires
+        /// "stale" proofs. What we COULD do is prevent a share delta if the timestamp is
+        /// too stale. 
+        // require(
+        //     oracleTimestamp + VERIFY_BALANCE_UPDATE_WINDOW_SECONDS >= block.timestamp,
+        //     "EigenPod.verifyBalanceUpdates: specified timestamp is too far in past"
+        // );
 
-        // Verify passed-in beaconStateRoot against oracle-provided block root:
-        BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot({
-            latestBlockRoot: eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp),
-            beaconStateRoot: stateRootProof.beaconStateRoot,
-            stateRootProof: stateRootProof.proof
-        });
+        // Check for a relevant partial withdrawal checkpoint using the oracle timestamp
+        // For balance updates, we update the current checkpoint only if the oracle timestamp
+        // is equal to the checkpoint timestamp
+        bool updateCheckpoint = currentCheckpointTimestamp == oracleTimestamp;
+        Checkpoint memory checkpoint = currentCheckpoint;
 
+        // Get the cached beacon state root for this timestamp, or validate `stateRootProof`
+        bytes32 beaconStateRoot = _getOrValidateStateRoot(oracleTimestamp, stateRootProof);
+
+        // Process each validator balance proof, keeping track of the total share delta
         int256 sharesDeltaGwei;
         for (uint256 i = 0; i < validatorIndices.length; i++) {
+            bytes32 pubkeyHash = validatorFields[i].getPubkeyHash();
+
             sharesDeltaGwei += _verifyBalanceUpdate(
                 oracleTimestamp,
+                pubkeyHash,
                 validatorIndices[i],
-                stateRootProof.beaconStateRoot,
+                beaconStateRoot,
                 validatorFieldsProofs[i], // Use validator fields proof because contains the effective balance
                 validatorFields[i]
             );
+
+            // If we're working on a partial withdrawal checkpoint and we haven't seen
+            // this validator yet, mark the validator as proven
+            if (updateCheckpoint && !validatorProven[oracleTimestamp][pubkeyHash]) {
+                validatorProven[oracleTimestamp][pubkeyHash] = true;
+                checkpoint.proofsRemaining--;
+            }
+        }
+
+        // If we have updated validator balances, send them to the EigenPodManager
+        if (sharesDeltaGwei != 0) {
+            eigenPodManager.recordBeaconChainETHBalanceUpdate(podOwner, sharesDeltaGwei * int256(GWEI_TO_WEI));
```

Comment:
we need to make sure that balance updates cannot be proven out of order

Reply:
Indeed - I modified `_verifyBalanceUpdate` to return a `sharesDeltaGwei` of 0 if `oracleTimestamp < validatorInfo.mostRecentBalanceUpdateTimestamp`.

So it's possible to submit balance proofs out of order, but they won't actually result in a share delta.

Reply:
ah didn't see that. great